src/pkg/filters/filters.go

```diff
@@ -359,21 +359,23 @@ func newSliceFilter(c comparator, targets, normTargets []string, negate bool) Fi
 // ----------------------------------------------------------------------------------------------------
 
 // CompareAny checks whether any one of all the provided
-// inputs passes the filter.
+// inputs passes the filter.  If one passes, that value is
+// returned, as well as its index in the input range.
+// If nothing matches, returns ("", -1, false)
 //
 // Note that, as a gotcha, CompareAny can resolve truthily
 // for both the standard and negated versions of a filter.
 // Ex: consider the input CompareAny(true, false), which
 // will return true for both Equals(true) and NotEquals(true),
 // because at least one element matches for both filters.
-func (f Filter) CompareAny(inputs ...string) bool {
-	for _, in := range inputs {
+func (f Filter) CompareAny(inputs ...string) (string, int, bool) {
```

Comment:
I'm very hesitant to return indexes. It seems like an excellent footgun as it makes the assumption that the caller has a stable slice/array for the input which is something very subtle that can be missed. For example, having something that converts an underlying map type to a slice to return either the keys or the values could return incorrect results if the following were used

```go
sites := // map of ids -> names
_, idx, matched := filter.CompareAny(maps.Values(sites)...)

// Get the "correct" site ID.
siteID := maps.Keys(sites)[idx]
```

Ideally someone would realize the fact that this wouldn't work, but it's something very subtle and can't be easily enforced by a linter. The compiler won't catch it either so it's pretty easy to miss

Reply:
nbd, will remove; it was an afterthought anyway.